lib/usd/translators/shading/usdUVTextureReader.cpp

```diff
@@ -0,0 +1,309 @@
+
+#include <maya/MDGModifier.h>
+#include <maya/MFnDependencyNode.h>
+#include <maya/MGlobal.h>
+#include <maya/MObject.h>
+#include <maya/MPlug.h>
+#include <maya/MStatus.h>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+class PxrMayaUsdUVTexture_Reader : public UsdMayaShaderReader {
+public:
+    PxrMayaUsdUVTexture_Reader(const UsdMayaPrimReaderArgs&);
+
+    bool Read(UsdMayaPrimReaderContext* context) override;
+
+    TfToken GetMayaNameForUsdAttrName(const TfToken& usdAttrName) override;
+};
+
+PXRUSDMAYA_REGISTER_SHADER_READER(UsdUVTexture, PxrMayaUsdUVTexture_Reader)
+
+TF_DEFINE_PRIVATE_TOKENS(
+    _tokens,
+
+    // Maya "file" node attribute names
+    (file)
+    (alphaGain)
+    (alphaOffset)
+    (colorGain)
+    (colorOffset)
+    (defaultColor)
+    (fileTextureName)
+    (outAlpha)
+    (outColor)
+    (outColorR)
+    (outColorG)
+    (outColorB)
+    (place2dTexture)
+    (coverage)
+    (translateFrame)
+    (rotateFrame)
+    (mirrorU)
+    (mirrorV)
+    (stagger)
+    (wrapU)
+    (wrapV)
+    (repeatUV)
+    (offset)
+    (rotateUV)
+    (noiseUV)
+    (vertexUvOne)
+    (vertexUvTwo)
+    (vertexUvThree)
+    (vertexCameraOne)
+
+    // UsdUVTexture Input Names
+    (bias)
+    (fallback)
+    (scale)
+    (wrapS)
+    (wrapT)
+
+    // Values for wrapS and wrapT
+    (black)
+    (repeat)
+
+    // UsdUVTexture Output Names
+    ((RGBOutputName, "rgb"))
+    ((RedOutputName, "r"))
+    ((GreenOutputName, "g"))
+    ((BlueOutputName, "b"))
+    ((AlphaOutputName, "a"))
+);
+
+static const TfTokenVector _Place2dTextureConnections = {
+    _tokens->coverage,
+    _tokens->translateFrame,
+    _tokens->rotateFrame,
+    _tokens->mirrorU,
+    _tokens->mirrorV,
+    _tokens->stagger,
+    _tokens->wrapU,
+    _tokens->wrapV,
+    _tokens->repeatUV,
+    _tokens->offset,
+    _tokens->rotateUV,
+    _tokens->noiseUV,
+    _tokens->vertexUvOne,
+    _tokens->vertexUvTwo,
+    _tokens->vertexUvThree,
+    _tokens->vertexCameraOne
+};
+
+PxrMayaUsdUVTexture_Reader::PxrMayaUsdUVTexture_Reader(const UsdMayaPrimReaderArgs& readArgs)
+    : UsdMayaShaderReader(readArgs)
+{
+}
+
+/* virtual */
+bool PxrMayaUsdUVTexture_Reader::Read(UsdMayaPrimReaderContext* context)
+{
+    const auto&    prim = _GetArgs().GetUsdPrim();
+    UsdShadeShader shaderSchema = UsdShadeShader(prim);
+    if (!shaderSchema) {
+        return false;
+    }
+
+    MStatus           status;
+    MFnDependencyNode depFn;
+    if (!(UsdMayaTranslatorUtil::CreateShaderNode(
+              MString(prim.GetName().GetText()),
+              _tokens->file.GetText(),
+              UsdMayaShadingNodeType::Texture,
+              &status,
+              &_mayaObject)
+          && depFn.setObject(_mayaObject))) {
+        // we need to make sure assumes those types are loaded..
+        TF_RUNTIME_ERROR(
+            "Could not create node of type '%s' for shader '%s'.\n",
+            _tokens->file.GetText(),
+            prim.GetName().GetText());
```

Comment:
Might want the full shader path here rather than just the name, and similarly elsewhere.